src/account/account.entity.ts

```diff
@@ -174,28 +104,65 @@ export class Account extends BaseEntity {
             }
         }
 
-        return new URL(
-            `.ghost/activitypub/${type}/${post.uuid}`,
-            `${Account.protocol}://${this.site!.host}`,
-        );
+        return new URL(`.ghost/activitypub/${type}/${post.uuid}`, this.apId);
     }
 
-    private static protocol: 'http' | 'https' =
-        process.env.NODE_ENV === 'testing' ? 'http' : 'https';
-
-    static createFromData(data: AccountData) {
-        return new Account(
-            data.id,
-            data.uuid,
-            data.username,
-            data.name,
-            data.bio,
-            data.avatarUrl,
-            data.bannerImageUrl,
-            data.site,
-            data.apId,
-            data.url,
-            data.apFollowers,
-        );
+    updateProfile(params: ProfileUpdateParams) {
+        type P = ProfileUpdateParams;
+        const get = <K extends keyof P>(prop: K): P[K] =>
```

Comment:
I just wanted to avoid the repetition of
```ts
username: params.username === undefined ? this.username : params.username,
```

But I can put it back if it's preferred

Reply:
Nah, this is a cool utility, maybe we have generic `utils` module thing somewhere?

Reply:
I did try to pull it out - but unfortunately it does't work well when you pull it out, because you need to pass in the `this` value and it becomes a bit of a mess like `get(this, params, 'key')` PLUS the types become super fucky